app/controllers/agents_controller.rb

```diff
@@ -208,6 +208,12 @@ def complete
     render json: @agent.complete_option(params[:attribute])
   end
 
+  def destroy_undefined
+    current_user.undefined_agents.destroy_all
```

Comment:
`undefined_agents` does not do the STI, it only returns an array of `Agent`s, I wish there would be nicer, per query, solution other then not selecting `agents.type`